marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -637,12 +638,30 @@ function findFlowsAsMap() {
   return flows;
 }
 
+function getRecordSize(docUri) {
+  const doc = cts.doc(docUri);
+  const sizes = ['B', 'KB', 'MB'];
+  const nodeKind = xdmp.nodeKind(doc.root);
+  let bytes = 0;
+
+  if(nodeKind === 'binary') {
+    bytes = xdmp.binarySize(fn.head(doc).root);
+  } else {
+    bytes = xdmp.binarySize(fn.head(xdmp.unquote(xdmp.quote(doc), null, "format-binary")).root)
+  }
+
+  let power = Math.floor(Math.log(bytes) / Math.log(1024));
+  power = power > 2 ? 2 : power;
+  return (bytes / Math.pow(1024, power)).toFixed(2) * 1 + ' ' + sizes[power];
```

Comment:
I didn't see any tests for record size? I recommend doing some unit tests here to verify you're getting the correct result. And instead of passing in a URI, pass in the document itself - that makes it easier to test, as you don't have to persist a document first. 

Should test documents that result in each of the 3 sizes being used too.